src/App.js

```diff
@@ -14,11 +14,58 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selected: 'Jam',
+  }
+
+  render() {
+    const { selected } = this.state;
+
+    return (
+      <>
+        {selected
+          ? (
+            <h1>
+              Selected good: -
+              {selected}
+
+              <button
+                type="button"
+                onClick={() => {
+                  this.setState({ selected: null });
+                }}
+              >
+                X
+              </button>
+            </h1>
+          ) : (
+            <h1>
+              No goods selected
+            </h1>
+          )}
+
+        {goodsFromServer.map(el => (
+          <>
+            <div>
+              {el}
+              {this.state.selected !== el && (
+                <button
+                  type="button"
+                  key={el}
```

Comment:
`key` should be on the `div` that rendered on the top level.
Actualy, it's better to use `ul`-`li` structure here. Semantically it's a list